src/Psalm/Internal/Analyzer/Statements/Block/SwitchCaseAnalyzer.php

```diff
@@ -98,10 +98,9 @@ public static function analyze(
             $case_context->inside_conditional = true;
 
             if (ExpressionAnalyzer::analyze($statements_analyzer, $case->cond, $case_context) === false) {
-                /** @psalm-suppress PossiblyNullPropertyAssignmentValue */
-                $case_scope->parent_context = null;
-                $case_context->case_scope = null;
-                $case_context->parent_context = null;
+                unset($case_scope->parent_context);
```

Comment:
I was certain it only applied to typed properties but I made a test and you're right, it emits a Warning in recent version of PHP (notice before PHP8).

However, I don't think it's an issue. Those variables are created locally and are not accessed after unset (mainly because there's a return just after it.)

Also those properties are not flagged nullable and Psalm was allowing them to be assigned to null(PHPStan raised that). I think it's just a memory tweak, the object will be destroyed shortly anyway